src/ray/raylet/node_manager.cc

```diff
@@ -2860,6 +2871,50 @@ void NodeManager::PublishInfeasibleTaskError(const RayTask &task) const {
   }
 }
 
+MemoryUsageRefreshCallback NodeManager::CreateMemoryUsageRefreshCallback() {
+  return [this](bool is_usage_above_threshold) {
```

Comment:
we need some inline comments on the policy we pick the worker to kill. also should we unit test this?

Reply:
(can be done later): we can have a separate component to pick a worker to kill.